lametro/views.py

```diff
@@ -585,11 +590,32 @@ def dispatch(self, request, *args, **kwargs):
 
         return response
 
+    def post(self, request, *args, **kwargs):
+        slug = self.kwargs['slug']
+        person = self.model.objects.get(slug=slug)
+        self.object = self.get_object()
+
+        # The submitted hidden field determines which form was used
+        if 'bio_form' in request.POST:
+            form = PersonBioForm(request.POST, instance=person)
+        elif 'headshot_form' in request.POST:
+            form = PersonHeadshotForm(request.POST, instance=person)
+
+        if form.is_valid():
+            form.save()
+            return HttpResponseRedirect(self.request.path_info)
+        else:
+            return self.render_to_response(self.get_context_data(form=form))
+
     def get_context_data(self, **kwargs):
 
         context = super().get_context_data(**kwargs)
         person = context['person']
 
+        # The first form is set with form_class, but we need to explicitly
+        # add the second, different form to the context.
+        context['form2'] = self.second_form_class
```

Comment:
How about adding both forms to the context here with descriptive names, e.g., headshot_form and biography_form?

Reply:
Done! That makes a lot more sense, I was following online suggestions a little too literally there.